src/device-api/index.ts

```diff
@@ -59,30 +57,16 @@ export class SupervisorAPI {
 			return res.sendStatus(200);
 		});
 
-		// Expires the supervisor's API key and generates a new one.
-		// It also communicates the new key to the balena API.
 		this.api.post(
 			'/v1/regenerate-api-key',
 			async (req: apiKeys.AuthorizedRequest, res) => {
-				await apiKeys.initialized();
-
-				// check if we're updating the cloud API key
-				const shouldUpdateCloudKey =
-					req.auth.apiKey === (await getGlobalApiKey());
-
-				// regenerate the key...
-				const newKey = await apiKeys.refreshKey(req.auth.apiKey);
-
-				// if we need to update the cloud API with our new key
-				if (shouldUpdateCloudKey) {
-					// report the new key to the cloud API
-					reportCurrentState({
-						api_secret: newKey,
-					});
+				const { apiKey: oldKey } = req.auth;
+				const newKey = await actions.regenerateKey(oldKey);
+				if (newKey === oldKey) {
+					// Regenerate failed, return old key
+					return res.status(500).send(oldKey);
```

Comment:
The `500` status is a new interface here to help differentiate whether a key refresh was successful or not.

Reply:
Not sure about returning data with a 500, server error just means server error right?

Reply:
True, it's not common behavior for a 500. The user should also already have the old key when requesting this endpoint, so I can remove this.

Reply:
Updated in 8410d18ce21014624c9cc14c1a64425c90f5f748